### DISCUSSION TITLE
Run in local environment

### DISCUSSION THREAD

**Discussion by codemouse83** [BoostryJP/ibet-Wallet-API#1043] [Q&A] [answered]
I read the Readme and would like to reproduce it in the local environment,
 but there are some unclear points and I would like to confirm it.

·Overall
Is the command execution environment in the Readme inside the container? Or is it a local PC (git clone environment)?

・ Starting and Stoppgin the Server
Is it correct to recognize that it is necessary to start the container with docker-compose up to execute ./bin_server.sh start?

・ Running the tests
pip install -r ./app/test/requirement.txt
Isn't it?

**Comment by codemouse83** [comment, marked as answer]
I'm not very familiar with it, but I think the readme needs work until you check the startup status of docker with docker-compose.

**Reply by YoshihitoAso** [reply]
Thank you for your feedback. We will try to improve the documentation based on the comments we received.

> ·Overall
> Is the command execution environment in the Readme inside the container? Or is it a local PC (git clone environment)?

The commands in the readme are for starting the server in a local environment.

> ・ Starting and Stoppgin the Server
> Is it correct to recognize that it is necessary to start the container with docker-compose up to execute ./bin_server.sh start?

docker-compose is for running unit tests on workflow. Please check the following.

https://github.com/BoostryJP/ibet-Wallet-API/blob/dev-21.9/.github/workflows/test.yml

When starting the service, you will need to set up GoQuorum, and deploy the contract, and set environment variables. Unfortunately, this is not fully supported in the documentation. We'll try to include this in the readme as soon as possible.

> ・ Running the tests
> pip install -r ./app/test/requirement.txt
> Isn't it?

To run unit tests in the local environment, install the package used for unit tests such as pytest and run pytest. Please check the following.

https://github.com/BoostryJP/ibet-Wallet-API#running-the-tests

**Reply by codemouse83** [reply]
Thank you for your answer.
I  confirm how to run program.

This is the test execution part,
```
pip install -r ./test/requirement.txt
```
is collect, isn't it?

I think it's a very interesting project, so I hope I can continue to be involved.

**Reply by YoshihitoAso** [reply]
Thank you

The old information was still in the Readme.
We have just corrected it.

https://github.com/BoostryJP/ibet-Wallet-API/commit/598b1c641712ded134ba48a484b5b4e080759b12
